central/notifications/service/service_test.go

```diff
@@ -0,0 +1,197 @@
+	result, err := s.service.CountNotifications(s.ctx, s.fakeCountNotificationsRequest)
+
+	s.ErrorIs(err, errFake)
+	s.Nil(result)
+}
+
+// Test GetNotification
+
+type getNotificationTestSuite struct {
+	baseTestSuite
+
+	fakeResourceByIDRequest     *v1.ResourceByID
+	fakeGetNotificationResponse *v1.GetNotificationResponse
+}
+
+func (s *getNotificationTestSuite) SetupTest() {
+	s.baseTestSuite.SetupTest()
+
+	s.fakeResourceByIDRequest = &v1.ResourceByID{Id: fakeID}
+	s.fakeGetNotificationResponse = &v1.GetNotificationResponse{
+		Notification: convertToServiceType(fakeNotification),
+	}
+}
+
+func (s *getNotificationTestSuite) TestGetNotification_Success() {
+	s.datastoreMock.EXPECT().GetNotificationByID(s.ctx, fakeID).Return(fakeNotification, nil)
+	result, err := s.service.GetNotification(s.ctx, s.fakeResourceByIDRequest)
+
+	s.Require().NoError(err)
+	s.Equal(s.fakeGetNotificationResponse, result)
+}
+
+func (s *getNotificationTestSuite) TestGetNotification_Error() {
+	s.datastoreMock.EXPECT().GetNotificationByID(s.ctx, fakeID).Return(nil, errFake)
+	result, err := s.service.GetNotification(s.ctx, s.fakeResourceByIDRequest)
+
+	s.ErrorIs(err, errFake)
+	s.Nil(result)
+}
+
+func (s *getNotificationTestSuite) TestGetNotification_NotFound() {
+	s.datastoreMock.EXPECT().GetNotificationByID(s.ctx, fakeID).Return(nil, errox.NotFound)
+	result, err := s.service.GetNotification(s.ctx, s.fakeResourceByIDRequest)
+
+	s.ErrorIs(err, errox.NotFound)
+	s.Nil(result)
+}
+
+// Test ListNotifications
+
+type listNotificationsTestSuite struct {
+	baseTestSuite
+
+	fakeStorageNotificationsList []*storage.Notification
+	fakeServiceNotificationsList []*v1.Notification
+	fakeListNotificationRequest  *v1.ListNotificationsRequest
+	fakeListNotificationResponse *v1.ListNotificationsResponse
+}
+
+func (s *listNotificationsTestSuite) SetupTest() {
+	s.baseTestSuite.SetupTest()
+
+	s.fakeStorageNotificationsList = []*storage.Notification{fakeNotification}
+	s.fakeServiceNotificationsList = []*v1.Notification{convertToServiceType(fakeNotification)}
+	s.fakeListNotificationRequest = &v1.ListNotificationsRequest{}
+	s.fakeListNotificationResponse = &v1.ListNotificationsResponse{Notifications: s.fakeServiceNotificationsList}
+}
+
+func (s *listNotificationsTestSuite) TestListNotifications_Success() {
+	s.datastoreMock.EXPECT().ListNotifications(s.ctx, gomock.Any()).Return(s.fakeStorageNotificationsList, nil)
+	result, err := s.service.ListNotifications(s.ctx, s.fakeListNotificationRequest)
+
+	s.Require().NoError(err)
+	s.Equal(s.fakeListNotificationResponse, result)
+}
+
+func (s *listNotificationsTestSuite) TestListNotifications_Error() {
+	s.datastoreMock.EXPECT().ListNotifications(s.ctx, gomock.Any()).Return(nil, errFake)
+	result, err := s.service.ListNotifications(s.ctx, s.fakeListNotificationRequest)
+
+	s.ErrorIs(err, errFake)
+	s.Nil(result)
+}
+
+// Test query builder
+
+func TestNotificationsQueryBuilder(t *testing.T) {
```

Comment:
I think we need a test(s) that:
* creates some test data in a test DB
* takes a bunch of filters as input (I don't see any filter validation, so I guess everything is fair game, including `nil`)
* verifies the responses make sense